app/js/lib/foxbox/foxbox.js

```diff
@@ -123,49 +127,66 @@ export default class Foxbox extends Service {
-          if (!Array.isArray(boxes) || boxes.length === 0) {
-            return resolve();
-          }
-
-          // We filter out boxes registered more than 5 minutes ago.
-          const now = Math.floor(Date.now() / 1000) - 60 * 5;
-          this[p.boxes] = Object.freeze(
-            boxes
-              .filter(box => box.timestamp - now >= 0)
-              .map(box => {
-                  // NOTE(sgiles): There is consideration to allow
-                  // only "local_origin" and "tunnel_origin", removing the need
-                  // to parse message - this merges the relevant message fields
-                  // into the main object
-                  const { local_origin, tunnel_origin } =
-                    JSON.parse(box.message);
-                  box.local_origin  = local_origin;
-                  box.tunnel_origin = tunnel_origin;
-                  return Object.freeze(box);
-              })
-          );
-
-          if (!this[p.settings].configured) {
-            this.selectBox();
-          }
-          resolve();
-        })
-        .catch(() => {
-          // When something goes wrong, we still want to resolve the promise so
-          // that a hostname set previously is reused.
-          resolve();
-        });
+    return new Promise(resolve => {
+      setTimeout(() => {
+        this[p.net].fetchJSON(this[p.settings].registrationService)
+          .then(boxes => {
+            if (!Array.isArray(boxes)) {
+              return resolve();
+            }
+
+            // We filter out boxes registered more than 2 minutes ago.
+            const now = Math.floor(Date.now() / 1000) - 60 * 2;
+            this[p.boxes] = Object.freeze(
+              boxes
+                .filter(box => box.timestamp - now >= 0)
+                .map(box => {
+                    // NOTE(sgiles): There is consideration to allow
+                    // only "local_origin" and "tunnel_origin", removing the
+                    // need to parse message - this merges the relevant message
+                    // fields into the main object
+                    const { local_origin, tunnel_origin } =
+                      JSON.parse(box.message);
+                    box.local_origin  = local_origin;
+                    box.tunnel_origin = tunnel_origin;
+                    return Object.freeze(box);
+                })
+            );
+
+            // If the registration server didn't give us any info and
+            // we have no record of previous registrations, we schedule
+            // a retry.
+            if (!this[p.boxes].length &&
+                !this[p.settings]._localOrigin &&
+                !this[p.settings]._remoteOrigin) {
+              console.warn('No boxes found. Retrying...');
+              return this._initDiscovery(1000);
+            }
+
+            if (!this[p.settings].configured) {
+              this.selectBox();
+            }
+            console.log('Box found');
```

Comment:
__nit:__ please don't forget to remove ```console.log``` from the final PR (I admit our log is already quite noisy even without this one :)), we'll switch to ```debug``` later). But if you think it's still useful let's make it more informative with something like ```console.log('Box found %s', ..url...of...the...box);``` (if possible).